CondTools/SiStrip/plugins/SiStripSummaryReader.cc

```diff
@@ -1,13 +1,34 @@
+// user include files
+#include "CondFormats/DataRecord/interface/SiStripSummaryRcd.h"
 #include "CondFormats/SiStripObjects/interface/SiStripSummary.h"
-
-#include "CondTools/SiStrip/plugins/SiStripSummaryReader.h"
-
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+// system include files
 #include <iostream>
 #include <string>
 #include <sstream>
 #include <cstdio>
 #include <sys/time.h>
 
+class SiStripSummaryReader : public edm::one::EDAnalyzer<> {
+public:
+  explicit SiStripSummaryReader(const edm::ParameterSet&);
+  ~SiStripSummaryReader() override = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  uint32_t printdebug_;
```

Comment:
shouldn't this be a bool after all?

Reply:
perhaps, what's the functional advantage?